sre-recipes/sandboxctl.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# -*- coding: utf-8 -*-
+"""
+This CLI is the interface for sre recipes. It deploys and rolls back
+broken services as well as verifies that the user found the correct
+cause of the broken service.
+
+For information on how to run the CLI, run the following:
+`python3 sandboxctl.py --help`
+"""
+
+import os
+from pkgutil import iter_modules
+from pathlib import Path
+from importlib import import_module
+from inspect import isclass
+import click
+from recipe import Recipe
+
+
+def get_recipes():
+    """
+    Gets all valid recipes in sre-recipes/recipes, and returns
+    the recipe names
+    """
+    recipes = {}
+    recipes_root_dir = os.path.dirname(os.path.abspath(__file__)) + '/recipes'
+    recipes_root_modules_path = Path(recipes_root_dir).resolve()
+    for (_, recipe_module_name, _) in iter_modules([recipes_root_modules_path]):
+        recipe_module_path = Path(recipes_root_dir + '/' + recipe_module_name).resolve()
+        for (_, recipe_name, _) in iter_modules([recipe_module_path]):
+            recipe_module = import_module('.'.join(['recipes', recipe_module_name, recipe_name]))
```

Comment:
I find this file parsing code above this hard to read. You should try to avoid changing between paths and strings so often, and make use of [Path's built-in `/` operator](https://stackoverflow.com/questions/48190959/how-do-i-append-a-string-to-a-path-in-python/48191073). You should also consider using something like `os.walk` to iterate files more cleanly

Many-nested loops like this aren't very pythonic in general though. You could replace 40-45 with a list comprehension:
```
from os.path import dirname, abspath

root_dir = Path(dirname(abspath(__file__))) / 'recipes'
modules = [f"recipes.{subdir}.{name}" 
                for subdir in os.listdir(root_dir)
                for _, name, _ in iter_modules([root_dir / subdir])]
```
That should give you a list of strings you can feed into `import_module`

Reply:
Thanks! I ended up converting as much as possible into list comprehensions.